components/clinical/plan-definition-editor/src/index.tsx

```diff
@@ -0,0 +1,240 @@
+    )
+
+    return {
+      problemActions: pa,
+      educationActions: ea,
+    }
+  }, [planDefinition.action])
+
+  // Calculate whether all problems in the given categories are enabled
+  const { allProblemsOn, allEducationOn } = useMemo(
+    () => ({
+      allProblemsOn:
+        problemActions?.map((pa) => pa?.elementId as string).every((id) => selectedProblemIds.includes(id)) ?? false,
+      allEducationOn:
+        educationActions?.map((ea) => ea?.elementId as string).every((id) => selectedProblemIds.includes(id)) ?? false,
+    }),
+    [educationActions, problemActions, selectedProblemIds]
+  )
+
+  const handleSingleActionChange: OnProblemChange = (e) => {
+    onProblemChange(e)
+  }
+
+  const handleToggleAllProblemsClick = (e: ChangeEvent<HTMLInputElement>) => {
+    const newState = problemActions?.map((p) => ({ problemId: p?.elementId as string, state: e.target.checked }))
+    newState && onProblemChange(newState)
+  }
+
+  const handleToggleAllEducationClick = (e: ChangeEvent<HTMLInputElement>) => {
+    const newState = educationActions?.map((p) => ({ problemId: p?.elementId as string, state: e.target.checked }))
+
+    newState && onProblemChange(newState)
+  }
+
+  return (
+    <>
+      <StyledSectionToggle>
+        <div>
+          <ToggleInput
+            checked={allProblemsOn}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => handleToggleAllProblemsClick(e)}
+            id="problem-toggle-all"
+          />
+        </div>
+        <label htmlFor="problem-toggle-all">Patient Problems</label>
+      </StyledSectionToggle>
+      <StyledProblemSection>
+        {problemActions?.map((problem, pidx) => {
+          const goal = planDefinition.goal?.find(
+            (g) => problem?.goalId && g?.elementId?.includes(problem.goalId[0] as string)
+          )
+          return (
+            <StyledListItem key={`problem-${pidx}`}>
+              <StyledProblemToggleRow>
+                <div>
+                  <ToggleInput
+                    checked={selectedProblemIds.includes(problem?.elementId as string)}
+                    onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                      handleSingleActionChange([{ problemId: problem?.elementId as string, state: e.target.checked }])
+                    }
+                    id={`problem-${pidx}`}
+                  />
+                </div>
+                <label htmlFor={`problem-${pidx}`}>{problem?.description}</label>
+              </StyledProblemToggleRow>
+              <div>
+                <StyledTargetRow>
+                  <span>
+                    <BullseyeIcon size="large" />
+                  </span>
+                  <div>{goal?.description?.text}</div>
+                </StyledTargetRow>
+
+                {problem?.action?.map((interventionPlan, ipidx) => (
+                  <StyledTargetRow key={`intervention-plan-${ipidx}`}>
+                    <span />

```

Comment:
Does this span serve a purpose?

Reply:
Only to remind me of my aging, forgetful brain. It's an artifact of refactoring into this library from ehr-client. Will remove, cheers :)